Sources/PushNotifications.swift

```diff
@@ -188,17 +201,23 @@ import UserNotifications
         return interests.filter { !self.validateInterestName($0) }
     }
 
+    #if os(iOS)
     private func registerForPushNotifications(application: UIApplication) {
         UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { (granted, error) in
             if (granted) {
                 DispatchQueue.main.async {
                     application.registerForRemoteNotifications()
                 }
             }
-
+            
             if let error = error {
                 print(error.localizedDescription)
             }
         }
     }
+    #elseif os(OSX)
+    private func registerForPushNotifications(application: NSApplication) {
+        application.registerForRemoteNotifications(matching: [.alert, .sound, .badge])
```

Comment:
More of a question than anything - are we assuming that every user will want to register for all of `[.alert, .sound, .badge]` notifications?

Reply:
If will be more flexible in the future. Users will be able to provide an array with options so we will register just specified notifications. It's on TODO 👍